src/app/app-routing.module.ts

```diff
@@ -1,21 +1,46 @@
 import { NgModule } from '@angular/core';
 import { Routes, RouterModule } from '@angular/router';
-import { CourseListModule } from './course-list/course-list.module';
-import { CoursesComponent } from './course-list/courses/courses.component';
 import { LoginPageComponent } from './login/login-page/login-page.component';
+import { PageNotFoundComponent } from './core/page-not-found/page-not-found.component';
+import { CoreModule } from './core/core.module';
+import { CoursesComponent } from './course-list/courses/courses.component';
 import { CourseFormComponent } from './course-list/course-form/course-form.component';
 import { EditCourseFormComponent } from './course-list/edit-course-form/edit-course-form.component';
+import { AuthGuardService as AuthGuard } from './services/auth-guard.service';
 
 const routes: Routes = [
-  { path: '', component: CoursesComponent },
-  { path: 'courses', component: CoursesComponent },
-  { path: 'login', component: LoginPageComponent },
-  { path: 'course-edit', component: CourseFormComponent },
-  { path: 'course-edit/:id', component: EditCourseFormComponent }
+  { path: '', redirectTo: 'courses', pathMatch: 'full' },
+  {
+    path: 'courses',
+    component: CoursesComponent,
+    canActivate: [AuthGuard]
+    /*children: [
+      { path: '', component: CoursesComponent, data: [{ breadcrumb: 'Courses'}] },
+      { path: 'new', component: CourseFormComponent, data: [{ breadcrumb: 'New Course'}] },
+      { path: ':id', component: EditCourseFormComponent, data: [{ breadcrumb: 'Edit Course'}] }
+    ]*/
+  },
+  {
+    path: 'courses/new',
+    component: CourseFormComponent,
+    data: { breadcrumb: 'New Course' },
+    canActivate: [AuthGuard]
+  },
+  {
+    path: 'courses/:id',
```

Comment:
Interesting case would be nice to handle - show 404 when there is no a course with typed id